test/exe/Main.hs

```diff
@@ -0,0 +1,141 @@
+-- Copyright (c) 2019-2020 The DAML and HLS Authors. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE CPP #-}
+
+module Main (main) where
+
+import Control.Applicative.Combinators
+import Control.Exception (catch)
+import Control.Monad
+import Control.Monad.IO.Class (liftIO)
+import Data.Char (toLower)
+import Data.Foldable
+import Data.List
+import Data.Rope.UTF16 (Rope)
+import qualified Data.Rope.UTF16 as Rope
+import Development.IDE.Core.PositionMapping (fromCurrent, toCurrent)
+import Development.IDE.GHC.Util
+import qualified Data.Text as T
+import Development.IDE.Spans.Common
+-- import Development.IDE.Test
+-- import Development.IDE.Test.Runfiles
+import Development.IDE.Types.Location
+import qualified Language.Haskell.LSP.Test as LSPTest
+import Language.Haskell.LSP.Test hiding (openDoc')
+import Language.Haskell.LSP.Types
+import Language.Haskell.LSP.Types.Capabilities
+import Language.Haskell.LSP.VFS (applyChange)
+import System.Environment.Blank (setEnv)
+import System.FilePath
+import System.IO.Extra
+import System.Directory
+import Test.QuickCheck
+import Test.QuickCheck.Instances ()
+import Test.Tasty
+import Test.Tasty.ExpectedFailure
+import Test.Tasty.HUnit
+import Test.Tasty.QuickCheck
+import Data.Maybe
+
+import TestUtils
+
+-- ---------------------------------------------------------------------
+
+main :: IO ()
+main = defaultMain $ testGroup "HLS"
+  [ testSession "open close" $ do
+      doc <- openDoc' "Testing.hs" "haskell" ""
+      void (message :: Session WorkDoneProgressCreateRequest)
+      void (message :: Session WorkDoneProgressBeginNotification)
+      closeDoc doc
+      void (message :: Session WorkDoneProgressEndNotification)
+  ]
+
+----------------------------------------------------------------------
+-- Utils
+
+
+testSession :: String -> Session () -> TestTree
+testSession name = testCase name . run
+
+{-
+testSessionWait :: String -> Session () -> TestTree
+testSessionWait name = testSession name .
+      -- Check that any diagnostics produced were already consumed by the test case.
+      --
+      -- If in future we add test cases where we don't care about checking the diagnostics,
+      -- this could move elsewhere.
+      --
+      -- Experimentally, 0.5s seems to be long enough to wait for any final diagnostics to appear.
+      ( >> expectNoMoreDiagnostics 0.5)
+
+pickActionWithTitle :: T.Text -> [CAResult] -> CodeAction
+pickActionWithTitle title actions = head
+  [ action
+  | CACodeAction action@CodeAction{ _title = actionTitle } <- actions
+  , title == actionTitle ]
+-}
+
+mkRange :: Int -> Int -> Int -> Int -> Range
+mkRange a b c d = Range (Position a b) (Position c d)
+
+run :: Session a -> IO a
+run s = withTempDir $ \dir -> do
+  let ghcideExe = hieCommand
+
+  -- Temporarily hack around https://github.com/mpickering/hie-bios/pull/56
```

Comment:
Do we still have to work around this? The pr is closed.

Reply:
I just copied this file as is from hie. If it is gone there, it can go here.

Reply:
Isnt that from ghcide?

Reply:
True. And unused, in favour of the the hspec stuff. I guess I can just delete that whole directory.